src/uct/cuda/cuda_copy/cuda_copy_ep.c

```diff
@@ -97,6 +98,40 @@ uct_cuda_copy_get_mem_type(uct_md_h md, void *address, size_t length)
     return mem_info.type;
 }
 
+static UCS_F_ALWAYS_INLINE ucs_status_t uct_cuda_copy_ctx_rsc_get(
+        uct_cuda_copy_iface_t *iface, uct_cuda_copy_ctx_rsc_t **ctx_rsc_p)
+{
+    CUcontext ctx;
+    ucs_status_t status;
+    khiter_t iter;
+    unsigned long long ctx_id;
+
+    status = UCT_CUDADRV_FUNC_LOG_ERR(cuCtxGetCurrent(&ctx));
+    if (ucs_unlikely(status != UCS_OK)) {
+        return status;
+    } else if (ucs_unlikely(ctx == NULL)) {
+        ucs_error("no context bound to calling thread");
+        return UCS_ERR_IO_ERROR;
+    }
+
+#if CUDA_VERSION >= 12000
+    status = UCT_CUDADRV_FUNC_LOG_ERR(cuCtxGetId(ctx, &ctx_id));
```

Comment:
From cuCtxGetId doc:
 Returns in ctxId the unique Id which is associated with a given context. The Id is unique for the life of the program for this instance of CUDA. **If context is supplied as NULL and there is one current, the Id of the current context is returned**.

So maybe we can avoid the extra call to cuCtxGetCurrent and pass NULL?

Reply:
We need a context to store it in the hash map value, thus we need to call `cuCtxGetCurrent` here.

Reply:
> We need a context to store it in the hash map value, thus we need to call `cuCtxGetCurrent` here.

Only in slow path when we initi ctx_rsc